control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -165,6 +166,24 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 	}
 
+	// Determine if active control plane operator image matches the expected release image payload. If it doesn't
+	// need to no-op until the operator is recreated and at the matching version
+	if _, ok := hostedControlPlane.Annotations[hyperv1.ActiveHypershiftOperatorImage]; !ok {
+		return ctrl.Result{}, fmt.Errorf("active hypershift operator image annotation not present")
+	}
+	pullSecret := common.PullSecret(hostedControlPlane.Namespace)
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(pullSecret), pullSecret); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to lookup pull secret: %w", err)
+	}
+	expectedImage, err := cpoutil.GetHypershiftComponentImage(ctx, hostedControlPlane.Annotations, hostedControlPlane.Spec.ReleaseImage, r.ReleaseProvider, hostedControlPlane.Annotations[hyperv1.ActiveHypershiftOperatorImage], pullSecret.Data[corev1.DockerConfigJsonKey])
```

Comment:
Why do we need to do this rather than just checking if our own image matches what is in the annotation?

Reply:
Not all HCPs specify the annotation: The algorithm is defined in the imageFetch support phase

Reply:
I can refactor to where basically that annotation always exists on the HCP: and it's not a direct copy of the one on the HostedCluster though and instead that is all done at hostedCluster time when setting the label on the HCP. And then the HCP effectively always has the label.

Reply:
done